### ISSUE TITLE
Virtual setup that represents multiple computers on same network

### ISSUE THREAD

**Issue by wigging** [irmen/Pyro5#100]
I have some Pyro5 code that communicates between two computers on my home network. The code works fine but I would like to test it with 5 to 10 computers on the same network. Since I don't have more than two physical computers, is there a way to test this virtually using virtual machines, Docker containers, or something else? Does Pyro5 work with any of these solutions?

**Comment by wigging** [comment]
I created a Docker image using the following Dockerfile:

```text
FROM python:3.13-slim

EXPOSE 5000

WORKDIR /usr/src/app

COPY . .

RUN pip install --upgrade pip \
    && pip install Pyro5

CMD ["/bin/bash"]
```

I build the image and run a container using the following commands:

```text
docker build --tag python313 .
docker run -it --rm -p 5000:5000 python313
```

The client code in the Docker container is:

```python
import Pyro5.api

def main():
    """Run the greeting client."""
    name = input("What is your name? ").strip()

    # Use name server object lookup to get response
    greeting_maker = Pyro5.api.Proxy("PYRONAME:example.greeting")
    response = greeting_maker.get_fortune(name)
    print(response)

if __name__ == "__main__":
    main()
```

The server code on the computer is:

```python
import Pyro5.api

@Pyro5.api.expose
class GreetingMaker(object):
    def get_fortune(self, name):
        line1 = f"Hello, {name}! Here is your fortune message:\n"
        line2 = "Tomorrow's lucky number is 12345678."
        return "\n" + line1 + line2

def main():
    """Run the greeting server."""

    # Make a pyro daemon and register greeting maker object
    daemon = Pyro5.api.Daemon(host="mac123", port=5000)
    uri = daemon.register(GreetingMaker)

    # Find the name server and register the object with a name
    ns = Pyro5.api.locate_ns()
    ns.register("example.greeting", uri)

    # Start the event loop of the server and wait for calls
    print("Ready.")
    daemon.requestLoop()

if __name__ == "__main__":
    main()
```

I ran the name server using `pyro5-ns -n mac123` but the client code in the Docker container is unable to reach the server running outside the container on the same computer. Any suggestions on how to get this to work?

**Comment by irmen** [comment]
Pyro5 just uses the basic TCP/IP network services.  In the documentation are some tips on what you might need to set in the configuration to make your servers externally reachable, but Pyro itself is not tied to one solution or another.  As long as the network setup properly ensures connectivity and port forwarding you should be able to configure Pyro clients and servers to talk to each other

**Comment by wigging** [comment]
> In the documentation are some tips on what you might need to set in the configuration to make your servers externally reachable

Is this the documentation you are referring to? https://pyro5.readthedocs.io/en/latest/config.html

**Comment by irmen** [comment]
Yes it is.  Also, the command line tools take certain options that are related to port and host configuration as well. These are described in the relevant chapters for example the name server https://pyro5.readthedocs.io/en/latest/nameserver.html#nameserver-nameserver
